obsplus/events/validate.py

```diff
@@ -100,57 +99,191 @@ def check_origins(event: Event):
+    """
+    if filt_amps is not None:
+        if type(filt_amps) is ResourceIdentifier:
+            filt_amps = filt_amps.id
+        bad = []
+        bad_filters = []
+        for amp in event.amplitudes:
+            if amp.filter_id.id != filt_amps:
+                wid = amp.waveform_id
+                nslc = (
+                    f"{wid.network_code}.{wid.station_code}."
+                    f"{wid.location_code}.{wid.channel_code}"
+                )
+                bad.append(nslc)
+                if amp.filter_id.id not in bad_filters:
+                    bad_filters.append(amp.filter_id.id)
+        assert len(bad) == 0, (
+            "Unexpected amplitude filter found:\n"
+            f"event_id: {event.resource_id.id}, "
+            f"seed_id/s: {bad}, "
+            f"filters_used: {set(bad_filters)}"
+        )
+
+
+@validator("obsplus", Event)
+def check_z_amps(event: Event, no_z_amps=False, **kwargs):
+    """
+    Check for IAML picks on Z channels (if no_z_amps is True)
+    """
+    if no_z_amps:
+        df = obsplus.picks_to_df(event)
+        df = df.loc[(df.evaluation_status != "rejected") & (df.phase_hint == "IAML")]
+        bad = df.loc[df.channel.str.endswith("Z")].seed_id.tolist()
+        assert len(bad) == 0, (
+            "Amplitude pick on Z axis found:\n"
+            f"event_id: {event.resource_id.id}, "
+            f"seed_id/s: {bad}"
+        )
+
+
+@validator("obsplus", Event)
+def check_amp_times(event: Event, **kwargs):
+    """
+    Check for amplitudes times that don't match the referenced pick time
     """
-    Perform tchecks on a events or event object.
+    bad = []
+    for amp in event.amplitudes:
+        if amp.time_window is None:
+            continue
+        amp_t = amp.time_window.reference
+        pick = amp.pick_id.get_referred_object()
+        if (amp_t is None) or (amp_t != pick.time):
+            wid = amp.waveform_id
+            nslc = (
+                f"{wid.network_code}.{wid.station_code}."
+                f"{wid.location_code}.{wid.channel_code}"
+            )
+            bad.append(nslc)
+    assert len(bad) == 0, (
+        "Mismatched amplitude and pick times found:\n"
+        f"event_id: {event.resource_id.id}, "
+        f"seed_id/s: {bad}, "
+    )
+
+
+def validate_catalog(events: Union[Catalog, Event], **kwargs) -> Union[Catalog, Event]:
+    """
+    Perform checks on a events or event object.
 
-    This function will try to fix any issues but will raise if it cannot.
+    This function will try to fix any issues but will raise if it cannot. It
+    is a simple wrapper around obsplus.validate.validate for the obsplus
+    namespace.
 
     Parameters
     ----------
     events
         The events or event to check
-
     """
-
     cat = events if isinstance(events, Catalog) else Catalog(events=[events])
-    for event in cat:
-        for func in CATALOG_VALIDATORS:
-            func(event)
+    validate(cat, "obsplus", **kwargs)
```

Comment:
Is it possible to turn certain of the validators in a particular namespace on/off? Also, how would one go about adding the checks defined above to other validators? (I'm sure the latter of these two is relatively doable, it just might be good to add an example to the docs for people who are not that familiar with the use of decorators.)

Reply:
> Is it possible to turn certain of the validators in a particular namespace on/off?

No, The idea is that one would create a namespace for a particular purpose. You can, however, control how much of the validators gets run using optional arguments in the validator function. You are right, the validators in the obsplus modules can be selected for new validation namespaces (like choosing items off a menu). I will add an example in the docs.